pkg/actuators/machine/actuator.go

```diff
@@ -79,13 +80,15 @@ func (a *Actuator) Create(ctx context.Context, machine *machinev1.Machine) error
 		awsClientBuilder: a.awsClientBuilder,
 	})
 	if err != nil {
-		return a.handleMachineError(machine, err, createEventAction)
+		fmtErr := fmt.Sprintf(scopeFailFmt, machine.GetName(), err)
+		return a.handleMachineError(machine, fmt.Errorf(fmtErr), createEventAction)
 	}
 	if err := newReconciler(scope).create(); err != nil {
 		if err := scope.patchMachine(); err != nil {
 			return err
 		}
-		return a.handleMachineError(machine, err, createEventAction)
+		fmtErr := fmt.Sprintf(reconcilerFailFmt, machine.GetName(), createEventAction, err)
```

Comment:
isn't this making us lose the error type machineapierros.InvalidMachineConfiguration MachineError?

Reply:
You are right. We are actually losing this type even before this formatting,
 - https://github.com/openshift/cluster-api-provider-aws/blob/master/pkg/actuators/machine/reconciler.go#L41 , I'll make a follow-up PR to fix it